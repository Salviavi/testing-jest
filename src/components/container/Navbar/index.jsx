import { Link } from "react-router-dom";

const Navbar = ({ onLogout }) => {
    const token = localStorage.getItem("access_token");

    return (
        <div className="navbar">
            <div className="navbar-links">
                <Link to="/">Home</Link>
            </div>
            <div className="navbar-buttons">
                {!token ? (
                    <>
                        <Link to="/login" className="login-button">Login</Link>
                        <Link to="/register" className="signup-button">Sign Up</Link>
                    </>
                ) : (
                    <button className="logout-button" onClick={onLogout}>Logout</button>
                )}
            </div>
        </div>
    );
}

export default Navbar;
