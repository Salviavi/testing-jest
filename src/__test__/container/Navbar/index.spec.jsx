import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "@/components/container/Navbar";

describe("Navbar Component", () => {
    it("should render Navbar with login and sign up links when no token", () => {
       // simulasi/mock token
        jest.spyOn(Storage.prototype, 'getItem').mockReturnValue(null);

        const { container, getByText } = render(
            <BrowserRouter>
                <Navbar />
            </BrowserRouter>
        );
        
        expect(getByText("Home")).toBeInTheDocument();
        expect(getByText("Login")).toBeInTheDocument();
        expect(getByText("Sign Up")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should render Navbar with logout button when token exists", () => {
        // Mock localStorage untuk simulasi access_token
        jest.spyOn(Storage.prototype, 'getItem').mockReturnValue('mock-token');

        const mockLogout = jest.fn();

        const { container, getByText } = render(
            <BrowserRouter>
                <Navbar onLogout={mockLogout} />
            </BrowserRouter>
        );

        expect(getByText("Logout")).toBeInTheDocument();
        fireEvent.click(getByText("Logout"));
        expect(mockLogout).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
    });
});
