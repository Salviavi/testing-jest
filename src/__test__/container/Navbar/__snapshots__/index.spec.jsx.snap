// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render Navbar with login and sign up links when no token 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navbar-links"
    >
      <a
        href="/"
      >
        Home
      </a>
    </div>
    <div
      class="navbar-buttons"
    >
      <a
        class="login-button"
        href="/login"
      >
        Login
      </a>
      <a
        class="signup-button"
        href="/register"
      >
        Sign Up
      </a>
    </div>
  </div>
</div>
`;

exports[`Navbar Component should render Navbar with logout button when token exists 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navbar-links"
    >
      <a
        href="/"
      >
        Home
      </a>
    </div>
    <div
      class="navbar-buttons"
    >
      <button
        class="logout-button"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
